src/services/member/plugins/action/service.ts

```diff
@@ -1,15 +1,70 @@
-import { CannotModifyOtherMembers } from '../../../../utils/errors';
+import { PermissionLevel } from '@graasp/sdk';
+
+import { CannotModifyOtherMembers, UnauthorizedMember } from '../../../../utils/errors';
 import { Repositories } from '../../../../utils/repositories';
 import { ActionService } from '../../../action/services/action';
+import { validatePermissionMany } from '../../../authorization';
+import { Item, ItemExtraMap } from '../../../item/entities/Item';
 import { Actor } from '../../entities/member';
 
+export const getMonthBeforeNow = () => {
+  const date = new Date(); // Today's date
+  date.setMonth(date.getMonth() - 1); // Set the date to one month ago
+  return date;
+};
+
 export class ActionMemberService {
   actionService: ActionService;
 
   constructor(actionService: ActionService) {
     this.actionService = actionService;
   }
 
+  async getFilteredActions(
+    actor: Actor,
+    repositories: Repositories,
+    filters: { startDate?: string; endDate?: string },
+  ) {
+    const { actionRepository } = repositories;
+
+    const { startDate, endDate } = filters;
+    const start = startDate ? new Date(startDate) : getMonthBeforeNow();
+    const end = endDate ? new Date(endDate) : new Date();
+
+    if (!actor) {
+      throw new UnauthorizedMember(actor);
+    }
+    const [actionsNeedsPermission, actionsWithoutPermssion] = await Promise.all([
+      actionRepository.getMemberActionsNeedPesrmission(actor.id, {
+        startDate: start,
+        endDate: end,
+      }),
+      actionRepository.getMemberActionsNoNeedForPesrmission(actor.id, {
+        startDate: start,
+        endDate: end,
+      }),
+    ]);
+
+    const setOfItemsToCheckPermission = Array.from(
+      new Map(actionsNeedsPermission.map(({ item }) => [item?.id, item])).values(),
+    );
```

Comment:
```suggestion
    const setOfItemsToCheckPermission = [
      ...new Set(actionsNeedsPermission.map(({ item }) => [item?.id, item])]
    
```
or similar

Reply:
set will not work with objects as every single object referring to a different reference, or a need to stringify item to use a set, Which I think doesn't really make difference than using `Map`